src/gui/torrentcontentmodel.cpp

```diff
@@ -113,31 +106,23 @@ namespace
         {
             const QString ext = info.suffix();
             if (!ext.isEmpty()) {
-                const QPixmap pixmap = pixmapForExtension(ext, QSize(32, 32));
-                if (!pixmap.isNull())
-                    return QIcon(pixmap);
+                auto cacheIter = m_iconCache.find(ext);
```

Comment:
@briankendall:  Can you add icon-caching to the Windows code above, as well?  

I think both platforms could benefit from this feature.